### ISSUE TITLE
Using workflow file from default branch

### ISSUE THREAD

**Issue by ZIJ** [diggerhq/digger#228]
From the user discussion @Spartakovic 

> ... when github triggers action on comments it’s using workflow file on the default branch, meaning it will use action version that is set on the default branch, which in turn might make it to use version that is different from your pull request action version

This seems to be unintuitive behavior. If a PR is raised and comment is added it's reasonable to expect that it will all be executed in the context of said branch. If for example I'm testing a workflow change that is only present in a particular circumstance, I'd make a branch with changes to reproduce and naturally would expect workflow file to work together with the code in the branch.

We may want to address it in fugure features; for now just capturing so it doesn't get lost @motatoes

**Comment by Spartakovic** [comment]
there is nothing we can do about it, it's how github works. We will need to completely change the way we trigger pipelines. Probably using github app and having backend or lambda

**Comment by ZIJ** [comment]
Indeed. The "backendful" approach with an orchestrating lambda or lightweight container already did feel kind of inevitable, we just didn't know why everyone else does it the hard way it was only reasonable to do it in a simple way until we face the critical mass of actual problems. This might be one of many such problems.

I'm adding a "need for backend" tag to track all related issues
